array_contains.json

```diff
@@ -138,5 +138,35 @@
             },
             "returns": false
         }
-    ]
+    ],
+    "process_graph": {
+        "find": {
+            "process_id": "array_find",
+            "arguments": {
+                "data": {
+                    "from_parameter": "data"
+                },
+                "value": {
+                    "from_parameter": "value"
+                }
+            }
+        },
+        "is_nodata": {
+            "process_id": "is_nodata",
+            "arguments": {
+                "x": {
+                    "from_node": "find"
+                }
+            }
+        },
+        "not": {
+            "process_id": "not",
+            "arguments": {
+                "x": {
+                    "from_node": "is_nodata"
+                }
+            },
+            "result": true
+        }
+    }
 }
```

Comment:
This implementation is inconsistent with the example `array_contains([1, 2, null], null) => true`

Reply:
There's a more general problem, too. We refer to eq, but eq on null always returns null.

Reply:
The question is how to best fix it:
1. array_find/contains can't search for null values
2. Rephrase `eq` to say "If exactly one of the operands is `null`, the return value is `null`." instead of "If any operand is `null`, the return value is `null`.". Would mean `eq(null, null)` => `true`, but `eq(null, 1)` => `null`. This would make array_find/contains work as expected.

Reply:
I'd go with 1) . `eq(null, null)` should return `null` (e.g. for booleans, eq is the as not(xor(null, null))) 

IMO `array_find` should not try and search for `null`s, but just return `null` if the searched value is `null`. If one really wanted to find the (first) index of a missing value, we could provide a more general process (`array_findfirst` or `array_select`) that takes a predicate (callback) and returns the first index where the predicate returns `true` - by using isValid in the predicate you can get the first `null`...

Reply:
On the other hand, you can also use eq in the property filters in load_collection for example and there null is not necessarily no-data and returning null could be somewhat unexpected.

Reply:
I think array_find and array_contains for null values can be emulated as follows:

array_find: `array_find(array_apply(data, x => is_nodata(x)), true)`
array_contains: `array_contains(array_apply(data, x => is_nodata(x)), true)`

Reply:
I'll leave this open for discussion until the end of the day. If there's no additional motion, I'll implement 1.

Reply:
Fine with #1

Reply:
For option 1 there's a PR here: #143, including examples for finding null values.
Do you agree, @mkadunc ?

Reply:
I agree with proposed changes